lib/p2p/Pool.ts

```diff
@@ -411,18 +413,50 @@ class Pool extends EventEmitter {
     if (!deal) {
       return;
     }
-    if (!deal.r_preimage) {
-      // TODO: Handle this unexpected case, if we are the maker we should have the preimage.
```

Comment:
Do we not want to keep this check? It seems like a possible edge case we should handle.

Reply:
deal.r_preimage is not used in this code at all. 

A check of it, while not used, looks out of order. And if it is undefined the `sendPaymentSync` will not give you the preimage and this is checked. So, not needed anymore

Reply:
My mistake, I thought it was being used.